templates/commands/render_test.go

```diff
@@ -86,7 +87,7 @@ func TestParseFlags(t *testing.T) {
 			t.Parallel()
 			r := &Render{}
 			err := r.parseFlags(tc.args)
-			if tc.wantErr != "" {
+			if err != nil || tc.wantErr != "" {
 				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
 					t.Fatal(diff)
 				}
```

Comment:
Sorry, I still don't understand why we need the preflight check on `err` or `tc.wantErr`. DiffErrString should be taking care of all of that.

Reply:
Consider the case where there's an err, and tc.wantErr matches it. Under your suggestion, DiffErrString would return "" and the test would continue on. It would call cmp.Diff to check the values of the flags, checks which we don't want (in this case).

Now you might be saying "yes we DO want to continue because we should then assert that the Render object is in a cleanly failed state, like just assert that it's a zero object or nil or something." But no, flag parsing will have left the the flags in a half-populated state that we should not predict or rely on.

Reply:
Ah I see. I think it's usually more obvious to do something like:

```go
if diff := testutils.DiffErrString(err, tc.wantErr); err != nil {
  t.Fatal(diff) // which will stop tests
}
```

or

```go
if diff := testutils.DiffErrString(err, tc.wantErr); err != nil {
  t.Error(diff)
}
if err != nil {
  return
}
```

Reply:
The first one looks not quite right (it will fail the test if there's an error, regardless of whether it was expected). I'll adopt the second one though.